packages/testsweets/lib/src/ui/widget_capture/widget_capture_viewmodel.dart

```diff
@@ -209,4 +212,86 @@ class WidgetCaptureViewModel extends FormViewModel {
     _activeWidgetId = '';
     captureWidgetStatusEnum = CaptureWidgetStatusEnum.inspectMode;
   }
+
+  bool _isEditMode = false;
+  bool get isEditMode => _isEditMode;
+
+  void editWidgetDescription() {
+    _isEditMode = true;
+    _widgetDescription = _activeWidgetDescription;
+    captureWidgetStatusEnum =
+        CaptureWidgetStatusEnum.captureModeWidgetNameInputShow;
+    notifyListeners();
+  }
+
+  void toggleIsEditMode() async {
+    _isEditMode = false;
+    try {
+      await _widgetCaptureService.loadWidgetDescriptionsForProject(
+        projectId: projectId,
+      );
+    } catch (e) {}
+    captureWidgetStatusEnum = CaptureWidgetStatusEnum.inspectMode;
+    notifyListeners();
+  }
+
+  Future<void> deleteWidgetDescription() async {
```

Comment:
I'd like to see unit tests that check
- If onChangeValue is empty we update the inputErrorMessage to `Widget name must not be empty`
- If not it's set to empty
- I want confirmation that we're calling the deleteWidgetDescription with the correct parameters
- I want a test that checks if the view is in edit mode when this function completes